gaeguli/record_target.c

```diff
@@ -0,0 +1,1006 @@
+/**
```

Comment:
I am very hesitant about the current record_target implementation because most parts are duplicated with _GaeguliTarget_. Strictly speaking, duplicated parts should be common codes.

Shouldn't we re-use _GaeguliTarget_ and add minimal codes that support recording?
Pipeline-wise, the only different part is to replace `srtsink` with `filesink`, and the other behaviors, like dynamically attaching to and detaching from `tee`, adjusting encoding parameters, are exactly same.

I'd like to avoid the way that I should maintain the same codes in different places by doubled-efforts.

Reply:
I agree. I would update the patch with reusing GaeguliTarget.